bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ClientLibraryAggregatorServiceImpl.java

```diff
@@ -36,113 +31,230 @@
-                if (resource != null && resource.getChild("clientlibs") != null) {
-                    Resource ampClientlib = resource.getChild("clientlibs/amp-clientlib");
-                    if (ampClientlib == null) {
+        }
+
+        try (ResourceResolver resourceResolver = getServiceResourceResolver()) {
+
+            // Iterate through each resource type and retrieve its clientlib categories.
+            for (String resourceType : resourceTypes) {
+
+                Resource resource = getResourceType(resourceResolver, resourceType);
+                if (resource == null) {
+                    continue;
+                }
+
+                // Resolve the resource type's clientlib.
+                Resource clientlib = null;
+                if (!primaryPathBlank) {
+                    clientlib = resource.getChild(primaryPath);
+                }
+                if (clientlib == null) {
+                    if (!fallbackPathBlank) {
+                        clientlib = resource.getChild(fallbackPath);
+                        if (clientlib == null) {
+                            continue;
+                        }
+                    } else {
                         continue;
                     }
-                    String[] componentCategories = ampClientlib.getValueMap().get("categories", String[].class);
-                    if (componentCategories != null) {
-                        categories.addAll(Arrays.asList(componentCategories));
-                    }
+                }
+
+                // Retrieve the resource type's clientlib categories.
+                String[] resourceCategories = clientlib.getValueMap().get(CATEGORIES, String[].class);
+                if (resourceCategories != null) {
+                    categories.addAll(Arrays.asList(resourceCategories));
                 }
             }
+        } catch (LoginException e) {
+            LOG.error("Unable to get the service resource resolver.");
         }
-        return getClientLibOutput(baseCategory, StringUtils.join(categories, ","), type);
+
+        return getClientLibOutput(StringUtils.join(categories, ","), type);
     }
 
-    private Resource getResource(ResourceResolver resourceResolver, String resourceType) {
-        String[] searchPaths = resourceResolver.getSearchPath();
-        for (String path : searchPaths) {
-            Resource resource = resourceResolver.getResource(path + resourceType);
+    /**
+     * Resolves the resource of a relative resource type path.
+     * @param resolver Provides search paths used to turn relative paths to full paths and resolves the resource.
+     * @param resourceType The resource type to resolve.
+     * @return The resource of the resource type path.
+     */
+    private Resource getResourceType(ResourceResolver resolver, String resourceType) {
+
+        Resource testResource = resolver.getResource(resourceType);
```

Comment:
what's this in here for? i'm assuming for testing but i don't know if this has an actual purpose